### PR TITLE
Represent unversioned.Time correctly in swagger spec

### LINKED ISSUE #2968: Use dateTime type in swagger
Swagger supports a `date-time` type:
https://github.com/swagger-api/swagger-spec/blob/master/versions/1.2.md

Our time fields are currently represented as strings.

It looks like go-restful does recognize time.Time as `date-time` (https://github.com/emicklei/go-restful/blob/master/swagger/model_builder.go), but we're using a wrapper.

Haven't investigated the best way to fix this yet.

### PR DESCRIPTION
**Pull request by mbohlool**
unversioned.Time previously represented as "string" type with no format while the correct representation should also have "date-time" as format. fixes #2968

### CODE REVIEW

File: pkg/genericapiserver/genericapiserver.go [hunk 1]
```diff
@@ -900,6 +900,14 @@ func (s *GenericAPIServer) InstallSwaggerAPI() {
 		ApiPath:         "/swaggerapi/",
 		SwaggerPath:     "/swaggerui/",
 		SwaggerFilePath: "/swagger-ui/",
+		SchemaFormatHandler: func(typeName string) string {
+			switch typeName {
+			case "unversioned.Time":
+			case "*unversioned.Time":
```
Comment (lavalamp) [inline pkg/genericapiserver/genericapiserver.go]: tiny, tiny nit: go lets you combine this as `case "a", "b":`...

**Comment by lavalamp** [comment]
LGTM, nit is not worth fixing. Thanks!

File: pkg/genericapiserver/genericapiserver.go (hunk 1 above)
Comment (mbohlool) [inline pkg/genericapiserver/genericapiserver.go]: That is actually a bug. "unversioned.Time" returns empty string. Fixed.

**Comment by lavalamp** [comment]
Right you are, I should have caught that. Thanks for noticing!

**Comment by lavalamp** [comment]
Looks like you need another run of hack/update-swagger-spec.sh after your fix.

**Pull request merged**